astroquery/cadc/core.py

```diff
@@ -0,0 +1,546 @@
+                All async methods should return the raw HTTP response.
+            """
+
+        request_payload = self._args_to_payload(coordinates=coordinates,
+                                                radius=radius,
+                                                collection=collection)
+        # primarily for debug purposes, but also useful if you want to send
+        # someone a URL linking directly to the data
+        if get_query_payload:
+            return request_payload
+        response = self.run_query(request_payload['query'], operation='sync')
+        return response
+
+    @class_or_instance
+    def query_name_async(self, name):
+        """
+        Query CADC metadata for a name and return the corresponding metadata in
+         the CAOM2 format (http://www.opencadc.org/caom2/).
+
+        Parameters
+        ----------
+        name: str
+                name of object to query for
+
+        Returns
+        -------
+        response : `astropy.Table`
+            Results of the query in a tabular format.
+
+        """
+        response = self.run_query(
+            "select * from caom2.Observation o join caom2.Plane p "
+            "on o.obsID=p.obsID where lower(target_name) like '%{}%'".
+            format(name.lower()), operation='sync')
+        return response
+
+    @class_or_instance
+    def get_data_urls(self, query_result, include_auxiliaries=False):
+        """
+        Function to map the results of a CADC query into URLs to
+        corresponding data that can be later downloaded.
+
+        The function uses the IVOA DataLink Service
+        (http://www.ivoa.net/documents/DataLink/) implemented at the CADC.
+        It works directly with the results produced by Cadc.query_region and
+        Cadc.query_name but in principle it can work with other query
+        results produced with the Cadc query as long as the results
+        contain the 'caomPublisherID' column. This column is part of the
+        caom2.Plane table.
+
+        Parameters
+        ----------
+        query_result : result returned by Cadc.query_region() or
+                    Cadc.query_name(). In general, the result of any
+                    CADC TAP query that contains the 'caomPublisherID' column
+                    can be use here.
+        include_auxiliaries : boolean
+                    True to return URLs to auxiliary files such as
+                    previews, False otherwise
+
+        Returns
+        -------
+        A list of URLs to data.
+        """
+
+        if not query_result:
+            raise AttributeError('Missing metadata argument')
+
+        try:
+            publisher_ids = query_result['caomPublisherID']
+        except KeyError:
+            raise AttributeError(
+                'caomPublisherID column missing from query_result argument')
+        result = []
+        for pid in publisher_ids:
+            response = requests.get(self.data_link_url, params={'ID': pid})
```

Comment:
```suggestion
            response = self._request('GET', self.data_link_url, params={'ID': pid})
```

Reply:
@keflavich if I make this change I get errors about missing self.cache_location and self._session. Am I not using the correct base class?

Reply:
This is the correct base class, but you need to call `super().__init__()` in the `__init__` method above, see e.g. https://github.com/astropy/astroquery/blob/master/astroquery/simbad/core.py#L265

Reply:
Doh! That must have been lost during the refactoring.